lib/query.c

```diff
@@ -470,11 +470,13 @@ static rpmdbMatchIterator initQueryIterator(QVA_t qva, rpmts ts, const char * ar
 
 	if (mi == NULL) {
 	    struct stat sb;
-	    if (lstat(fn, &sb) != 0)
+	    char * full_fn = rpmGenPath(rpmtsRootDir(ts), fn, NULL);
```

Comment:
How about using the simpler `rpmGetPath()` instead? The Gen variant only differs in that it also accepts URLs which I don't think we need here :smile: